cmd/keymasterd/certgen.go

```diff
@@ -238,15 +266,20 @@ func (state *RuntimeState) getUserGroups(username string) ([]string, error) {
 		if err != nil {
 			continue
 		}
-		return groups, nil
+		return true, prependGroups(groups, ldapConfig.GroupPrepend), nil
 
 	}
-	if ldapConfig.LDAPTargetURLs == "" {
-		var emptyGroup []string
-		return emptyGroup, nil
+	return true, nil, errors.New("error getting the groups")
+}
+
+func (state *RuntimeState) getUserGroups(username string) ([]string, error) {
+	if config, groups, err := state.getLdapUserGroups(username); config {
```

Comment:
you changed the internal api here. before on empty group configuration we return an empty list and no error, now we return the nil. Would this break anything?

Reply:
No. I checked. Also, empty slices and nil slices are effectively the same.